[Follow-up] Meetup with Techworld with Nana SF meetup (#3563)
pulumi/pulumi-hugo: themes/default/content/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/index.md

# the time portion of the date value; posts are sorted in descending order by
# date/time.
date: 2023-11-13T22:05:34Z

# The draft setting determines whether a post is published. Set it to true if
# you want to be able to merge the post without publishing it.
draft: false

# Use the meta_desc property to provide a brief summary (one or two sentences)
# of the content of the post, which is useful for targeting search results or
# social-media previews. This field is required or the build will fail the
# linter test. Max length is 160 characters.
meta_desc: AI-driven Development is Here - Watch the discussion on AI in DevOps and how AI will impact the future of DevOps and Software Engineering.

# The meta_image appears in social-media previews and on the blog home page. A
# placeholder image representing the recommended format, dimensions and aspect
# ratio has been provided for you.
meta_image: ai-in-devops-meta-image.png

# At least one author is required. The values in this list correspond with the
# `id` properties of the team member files at /data/team/team. Create a file for
# yourself if you don't already have one.
authors:
    - sara-huddleston

>     - scott-lowe

Reviewer: You don't need to put my name on here. :-)

# At least one tag is required. Lowercase, hyphen-delimited is recommended.
tags:
    - devops
    - pulumi-events
    - ai
    - ml
    - community
    - platform_engineering

# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md
# for details, and please remove these comments before submitting for review.
---

> The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. This article unwraps the topics addressed during our “AI: Friends or For | AI Talks for DevOps” event in San Francisco.

Reviewer: ```suggestion
The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. This article unwraps the topics addressed during our “AI: Friend or Foe | AI Talks for DevOps” event in San Francisco.
```

> The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. This article unwraps the topics addressed during our “AI: Friends or For | AI Talks for DevOps” event in San Francisco.

Author: ```suggestion
The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. This article unwraps the topics addressed during our “AI: Friends or Foe | AI Talks for DevOps” event in San Francisco.
```

> The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. The biggest problem is trust. This article unwraps the topics addressed during our “AI: Friends or For | AI Talks for DevOps” event in San Francisco.

Reviewer: ```suggestion
The integration of artificial intelligence (AI) with DevOps signals a new era in software development. DevOps possesses unique characteristics and needs that make it exceptionally compatible with AI augmentation. Given that code fundamentally relies on language, and large language models (LLMs) serve as the core of GPT functionality, these models are particularly well-suited for tasks such as code generation. The biggest problem is trust. This article unwraps the topics addressed during our “AI: Friend or Foe | AI Talks for DevOps” event in San Francisco.
```

<!--more-->

> This article unwraps the topics addressed during our “AI: Friends or For | AI Talks for DevOps” event in San Francisco. Watch the discussion between Luke Hoban, Pulumi's CTO and cofounder of TypeScript, and Nana from TechWorld of Nana, moderated by community engineer Adora Nwodo, on How AI will impact the Future of DevOps.

Reviewer: ```suggestion
Watch the discussion between Luke Hoban, Pulumi's CTO and cofounder of TypeScript, and Nana from TechWorld of Nana, moderated by community engineer Adora Nwodo.
```

## On this AI in DevOps article:

- [How will AI Impact the Future of DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-will-ai-impact-the-future-of-devops)
- [What is DevOps in AI](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#what-is-devops-in-ai)
- [What is AI in DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#what-is-ai-in-devops)
- [How AI is Currently Used in DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-ai-is-currently-used-in-devops)

> - [How can a DevOps team take advantage of Artificial Intelligence](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-can-a-devops-team-take-advantage-of-artificial-intelligence)

Author: ```suggestion
- [How Can a DevOps Team Take Advantage of Artificial Intelligence](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-can-a-devops-team-take-advantage-of-artificial-intelligence)
```

> - [How will AI Impact the Future of DevOps](https://www.pulumi.com/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-will-ai-impact-the-future-of-devops)
> - [What is DevOps in AI](/#what-is-devops-in-ai)
> - [What is AI in DevOps](/#what-is-ai-in-devops)
> - [How AI is Currently Used in DevOps](/#how-ai-is-currently-used-in-devops)
> - [How can a DevOps team take advantage of Artificial Intelligence](/#how-can-a-devops-team-take-advantage-of-artificial-intelligence)
> - [Top 5 Skills for DevOps Engineers in the AI era](/#top-5-skills-for-devops-engineers-in-the-ai-era)
> - [How to Overcome AI for DevOps Challenges](/#how-to-overcome-ai-for-devops-challenges)
> - [Frequently Asked Questions](/#frequently-asked-questions)

Author: ```suggestion
- [How will AI Impact the Future of DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-will-ai-impact-the-future-of-devops)
- [What is DevOps in AI](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#what-is-devops-in-ai)
- [What is AI in DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#what-is-ai-in-devops)
- [How AI is Currently Used in DevOps](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-ai-is-currently-used-in-devops)
- [How can a DevOps team take advantage of Artificial Intelligence](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-can-a-devops-team-take-advantage-of-artificial-intelligence)
- [Top 5 Skills for DevOps Engineers in the AI era](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#top-5-skills-for-devops-engineers-in-the-ai-era)
- [How to Overcome AI for DevOps Challenges](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#how-to-overcome-ai-for-devops-challenges)
- [Frequently Asked Questions](/blog/devops-ai-developer-future--pulumi-user-group-tech-talks/#frequently-asked-questions)
```

## How will AI Impact the Future of DevOps

In the Fireside chat, Luke Hoban expressed his enthusiasm for incorporating AI into DevOps, emphasizing its potential utilization by both DevOps professionals and the development team. In contrast, Nana Janashia, [TechWorld with Nana](https://www.youtube.com/@TechWorldwithNana), adopted a more cautious stance, drawing attention to potential risks, particularly regarding reliability.

{{< youtube "i09F14yc0l4?rel=0" >}}

## What is DevOps in AI

When considering the adoption of AI and AI engineering, such as prompt engineering and other specific disciplines, AI engineers rely heavily on the cloud. Many AI-based systems are built around cloud-native tools, practices, and managed services.

> The extensive use of Cloud-native Primitives means that AI is fundamentally a cloud adoption story. Consequently, established DevOps and Infrastructure as Code tooling and best practices play an important role in AI engineering, in what is being done, and in leveraging its use.

Reviewer: ```suggestion
The extensive use of cloud-native primitives means that AI is fundamentally a cloud adoption story. Consequently, established DevOps and infrastructure as code tooling and best practices play an important role in AI engineering, in what is being done, and in leveraging its use.
```

## What is AI in DevOps

Integrating AI in DevOps entails employing machine learning (ML) and other artificial intelligence technologies to automate and optimize various software development and delivery aspects, including tasks such as automating testing and deployment processes and refining resource management.

The incorporation of AI in DevOps can result in:

- Enhanced speed,
- Improved resource management,

> - Increased Reliability throughout the software development lifecycle.

Reviewer 2: ```
- Enhanced speed
- Improved resource management
- Increased Reliability throughout the software development lifecycle
```

Remove end of line punctuation from list

> - Enhanced speed,
> - Improved resource management,
> - Increased Reliability throughout the software development lifecycle.

Author: ```suggestion
- Enhanced speed
- Improved resource management
- Increased Reliability throughout the software development lifecycle
```

> - Increased Reliability throughout the software development lifecycle.

Reviewer: ```suggestion
- Increased reliability throughout the software development lifecycle.
```

> - Increased Reliability throughout the software development lifecycle

Reviewer: ```suggestion
- Increased reliability throughout the software development lifecycle
```

Consequently, organizations may experience swifter deployments, reduced errors, and heightened overall productivity. By substituting manual processes with automation and AI-powered solutions, DevOps teams can elevate product quality and more efficiently oversee their systems.

## How AI is Currently Used in DevOps

{{< figure alt="Luke Hoban at the San Francisco Pulumi Group meetup discussing the usage of AI, during the AI Talks for DevOps" src="./luke-nana-adora-discussing-devop-ai.png" caption="Luke Hoban at the San Francisco Pulumi Group meetup discussing the usage of AI, during the AI Talks for DevOps" width=100% >}}

DevOps significantly impacts all stages of the development and operations lifecycle. It has helped break silos and optimize to increase developer velocity and productivity, creating a positive impact on the ops side.

AI assists DevOps teams to refine their operations by detecting inefficiencies and triggering warnings as soon as issues appear. This technology also enables real-time observation of systems and applications, permitting prompt actions from operations personnel and minimizing downtime.

> AI-driven automation can also help the software development process with code generation. The Operations teams from both departments quickly pinpoint any issues in their workflows while speeding up decision-making processes.

Reviewer: ```suggestion
AI-driven automation can also help the software development process with code generation. The operations teams from both departments quickly pinpoint any issues in their workflows while speeding up decision-making processes.
```

> ***“In the near term, AI usage in DevOps is about how do you keep the human in the loop, how do you use it as a learning tool, as a code generation tool, and for auditing to provide actionable notifications, all of which are still very human-dependent. There is so much more it can be used for. Currently, it's a tool serving humans. But it is fascinating to imagine it in the future when it can be trusted to be an autonomous agent.”*** - *Luke Hoban, CTO of Pulumi and co-creator of TypeScript*

So, how does this technology bring about these advancements? Two main ways are by streamlining development tasks and improving monitoring and security protocols.

### Streamlining Development Workflows

AI has the potential to significantly change the software development life cycle by automating routine processes, using resources efficiently, and offering solutions for improved code. DevOps teams can leverage AI-driven tools to increase effectiveness when managing their assets while also improving code quality, leading to faster project deliveries.

> For instance, Artificial Intelligence can identify any slowdowns during progress and recommend improvements, analyze root causes, and help make judicious decisions regarding resource allocation, ultimately resulting in efficient delivery of software products.

Reviewer: ```suggestion
For instance, AI can identify any slowdowns during progress and recommend improvements, analyze root causes, and help make judicious decisions regarding resource allocation, ultimately resulting in efficient delivery of software products.
```

### Enhancing Monitoring and Security

The usage of AI significantly boosts DevOps monitoring and security. With the assistance of AI tools, operations teams can use machine learning algorithms to examine a massive amount of data to protect their applications from potential risks and enhance reliability. These algorithmic solutions provide valuable advice for development personnel, which helps strengthen protection measures while optimizing performance simultaneously.

> ## How can a DevOps team take advantage of Artificial Intelligence

Reviewer: ```suggestion
## How can a DevOps team take advantage of AI
```

> ## How can a DevOps team take advantage of AI

Author: ```suggestion
## How Can a DevOps Team Take Advantage of Artificial Intelligence
```

> A DevOps team can leverage Artificial Intelligence (AI) in several ways to enhance efficiency, automation, and overall productivity. Here are some key strategies for a DevOps team to leverage AI:

Reviewer: ```suggestion
A DevOps team can leverage AI in several ways to enhance efficiency, automation, and overall productivity. Here are some key strategies for a DevOps team to leverage AI:
```

> A DevOps team can leverage Artificial Intelligence (AI) in several ways to enhance efficiency, automation, and overall productivity. Here are some key strategies for a DevOps team to leverage AI:

Reviewer: ```suggestion
A DevOps team can leverage AI in several ways to enhance efficiency, automation, and overall productivity. Here are some key strategies for a DevOps team to leverage AI:
```

- show me all s3 buckets not tagged in production
- show azure and azure native security groups
- show all AWS VPCs

### Remediation tool: Continuous and Automatic Compliance

Defining and managing security rules, criteria, and conditions are key to enabling security at scale. Policy as Code, or Compliance as Code, has the ability to verify and spot problems before deploying the infrastructure. When policies are written with code, you can apply software development practices such as testing, automated deployment, and version control.

AI can be valuable not just in anomaly detection but also in remediation, which doesn't just warn or error but can change the infrastructure directly to remediate compliance gaps. With Pulumi, [Remediation policies](https://www.pulumi.com/blog/remediation-policies/) can be configured and applied across any subset of stacks within an organization.

Consider the [unfortunate incident at GitLab](https://techcrunch.com/2017/02/01/gitlab-suffers-major-backup-failure-after-data-deletion-incident/), marked by a disastrous sequence of events resulting in the loss of segments of production databases and substantial user data.

> {{< youtube "tLdRBsuvVKc&t?rel=0" >}}

Author: ```suggestion
{{< youtube "tLdRBsuvVKc?si=brUXETRY4Rew7954?rel=0" >}}
```
